Sources/Yams/Constructor.swift

```diff
@@ -128,7 +128,7 @@ extension Date: ScalarConstructible {
             let length = $0.count
             let nanosecond: Int?
             if length < 9 {
-                nanosecond = Int($0 + String(repeating: "0", count: 9 - length))
+                nanosecond = Int($0).map { repeatElement(10, count: 9 - length).reduce($0, *) }
```

Comment:
I'm sorry I don't understand what this is doing and how this patch changes behavior. Could you explain please?

Reply:
I think part of what contributes to my confusion is the shorthand closure param identifier `$0` being used in a pretty complex closure with nested closures.